lib/validator.js

```diff
@@ -0,0 +1,15 @@
+'use strict';
+
+const Joi = require('joi');
+
+module.exports = {
+  auth: Joi.object().keys({
+    credentials: Joi.object().required()
+  }),
```

Comment:
The `onPreResponse` extension point is reached even if there's an authentication error, but I realized that the check for `request.response.source` would prevent the plugin from running when there's an authentication error. And you bring up a good point about projects that don't care about authentication. So I'll remove this validation and its tests.